.github/workflows/code-quality-analysis.yml

```
@@ -0,0 +1,101 @@
+        python scripts/advanced_quality_analysis.py
+
+    - name: Upload quality reports
+      uses: actions/upload-artifact@v4
+      with:
+        name: quality-reports
+        path: test_reports/
+        retention-days: 30
+
+    - name: Comment PR with quality results
+      if: github.event_name == 'pull_request'
+      uses: actions/github-script@v6
```

actionlint (1.7.7)

the runner of "actions/github-script@v6" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)